instaclustr/resource_cluster.go

```diff
@@ -23,6 +23,7 @@ var (
 		"RUNNING":     true,
 		"PROVISIONED": true,
 	}
+	semanticVersioningPattern, _ = regexp.Compile("[0-9]+\\.[0-9]+(\\.[0-9]+)*")
```

Comment:
ok, so this regex allows 2+ versions. so 3.1 is valid, and so is 3.1.2.3.4.5.6.7.7
Is that intentional?
(If so, could be simplified to [0-9]+(\\.[0-9]+)+

Reply:
So that last * should be a ?, which will only match duplets or triplets. Since this is always a left-most retrieval any subsequent matched of duplets and triplets are ignored.